libs/ml/tests/ml/training/basic_training.cpp

```diff
@@ -113,11 +116,12 @@ void PlusOneTest()
     cur_input.At(0, 0) = data.At(step, 0);
     g.SetInput(input_name, cur_input);
     cur_gt.At(0, 0) = gt.At(step, 0);
+    g.SetInput(label_name, cur_gt);
 
-    auto results = g.Evaluate(output_name);
-    loss += criterion.Forward({results, cur_gt});
-
-    g.BackPropagate(output_name, criterion.Backward({results, cur_gt}));
+    auto error_tensor = g.Evaluate(error_name);
```

Comment:
can the loss ops not be made batch aware so that the user doesn't have to manually compute the mean of the loss?